File: src/main/java/io/github/oleksivio/telegram/bot/api/TelegramConfigurationReference.java
package io.github.oleksivio.telegram.bot.api;

import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;

@Configuration
@ComponentScan({
        "ru.ioleksiv.telegram.bot.core.controller",
        "ru.ioleksiv.telegram.bot.core.configuration",
})
public class TelegramConfigurationReference {
}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/TelegramConfigurationReference.kt
package io.github.oleksivio.telegram.bot.api

import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration

/**
 * Must be added into Spring `@SpringBootApplication` class
 *
 * Example:
 *
 * ```kotlin
 * @SpringBootApplication
 * @EnableScheduling
 * @Import(TelegramConfigurationReference::class)
 * class Application
 * fun main(args: Array<String>) {
 *     runApplication<Application>(*args)
 * }
 * ```
 */
@Configuration
@ComponentScan(
        "io.github.oleksivio.telegram.bot.core.controller",
        "io.github.oleksivio.telegram.bot.core.configuration"
)
class TelegramConfigurationReference

